torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -123,6 +124,31 @@ std::string cudaDtypeCppString(const Dtype& dtype) {
   }
 }
 
+void CudaAnalysis::visit(const Free* v) {
+  if (thread_local_bufs_.count(v->buffer_var()) == 0 &&
+      cross_block_bufs_.count(v->buffer_var()) == 0) {
+    throw std::runtime_error("Global free not supported yet");
+  }
+}
+
+void CudaAnalysis::visit(const Allocate* v) {
+  Stmt* p = v->get_parent();
+  while (p) {
+    const For* for_v = dynamic_cast<const For*>(p);
+    if (for_v) {
+      if (for_v->loop_options().is_gpu_block_index()) {
+        cross_block_bufs_.insert(v->buffer_var());
```

Comment:
Minor: how about the case where thread_idx is outside block_idx? It is okay to ignore this for now. But maybe a TODO to remind ourselves of this case down the road.
```
for i in (0..10):   # theadIdx
   for j in (0..100):  # blockIdx
      t1 = alloc(1)
      ....
```

Reply:
Good point, this is an important case. I'll come back to it.